app/assets/javascripts/Components/Modals/graders_distribution_modal.jsx

```diff
@@ -0,0 +1,73 @@
+import React from "react";
+import Modal from "react-modal";
+import PropTypes from "prop-types";
+
+export class GraderDistributionModal extends React.Component {
+  static defaultProps = {
+    override: false,
+  };
+
+  constructor(props) {
+    super(props);
+    this.state = {
+      override: this.props.override,
+    };
+  }
+
+  componentDidMount() {
+    this.props.graders.forEach(grader => {
+      !this.props.weightings[grader._id] ? (this.props.weightings[grader._id] = 1) : null;
```

Comment:
Okay so overall, you shouldn't mutate `this.props` in a React component (a component should treat its props as immutable). While one alternative would be to store mutable data in `this.state`, in this particular case I don't think that's necessary.

We don't need to store the weights at all---instead, in the `onSubmit` function you can create an object that maps the grader ids to weights, and then pass that object to `this.props.onSubmit` as an argument. (Rather than accessing the props from the `onSubmit` callback function in the parent component.)

Reply:
@WeissSa you can do this by using plain Javascript to interact with the input elements. The easiest way to set it up in this case is to use a **ref** to refer to the form itself, then go through its input values.

See:

- https://reactjs.org/docs/uncontrolled-components.html
- https://developer.mozilla.org/en-US/docs/Web/API/FormData/FormData